docs/changes/coordinates/16246.feature.rst

```diff
@@ -0,0 +1,9 @@
+The ``SkyCoord`` and ``BaseCoordinateFrame`` ``separation()`` methods now
```

Comment:
It would maybe feel more natural to rephrase this changelog entry as "A warning is now emitted by default in case ... . It is also possible to suppress this warning or to promote it to an error via the new keyword argument ...", instead of introducing the new argument first.